docs/user-event/intro.mdx

```diff
@@ -0,0 +1,46 @@
+---
+id: intro
+title: Introduction
+---
+
+[`user-event`](https://github.com/testing-library/user-event)
+is a companion library for Testing Library that simulates user interactions by
+dispatching the events that would happen if the interaction took place in a browser.
+
+While most examples with `user-event` are for `React`, the library can be used
+with any framework as long as there is a DOM.
+
+## Difference to `fireEvent`
+
+The built-in [`fireEvent`](dom-testing-library/api-events.mdx#fireevent) is a
+utility to easier dispatch events.  
+It dispatches exactly the event you tell it to and just those - even if that
+exact event never had been dispatched in a real interaction in a browser.
+
+`user-event` on the other hand dispatches the events like they would happen if a
+user interacted with the document.  
+That might lead to the same event you previously dispatched per `fireEvent`
+directly, but it also might catch bugs that make it impossible for a user to
+trigger said event.  
+This is
+[why you should use `user-event`](https://ph-fritsche.github.io/blog/post/why-userevent)
+to test interaction with your components.
+
+## Writing tests with `userEvent`
+
+We recommend to use [`userEvent.setup()`](setup) when rendering your component
+and inline that rendering and setup in your test or use a setup function.  
+We discourage rendering or using any `userEvent` functions outside of the test
+itself - e.g. in a `before`/`after` hook - for reasons described in
+["Avoid Nesting When You're Testing"](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing).
+
+```js
+test('trigger some awesome feature when clicking the button', async () => {
+    const user = userEvent.setup()
```

Comment:
This is only my opinion, but is there a way to `setup` userEvent in the `setupFilesAfterEnv` in jest for example?
I think causing our users to write this line in every test might cause some pain.

Reply:
Yes, this could be done in a `beforeEach`. But we especially mention [Avoid nesting when you're testing](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing) in the docs...

The solution for now would be:
```js
function setup(jsx) {
  render(jsx)
  const user = userEvent.setup()
}

test('something', () => {
  const { user } = setup(<MyComponent/>)
  // ...
})
```

This pattern might even become *that* common that we should consider returning the user-event instance from `render` in the framework packages.

Reply:
Added an example for a test setup function below.

Reply:
I'm in favor of returning the user-event instance from `render`. 
But this means people will need to install user-event too and the frameworks will instantiate that for them?
That sounds too "voodoo" to me.

Reply:
Yes, something like that.

The framework wrapper could `try..require` `user-event`. So that if `user-event` is installed the wrapper will automatically return an instance from `render`.

Or we could take a similar approach like how `@testing-library/react` configures `/dom` when it's imported.
`user-event` could register itself on the framework wrapper when being imported.